### PR TITLE
Better command line in testeth

### PR DESCRIPTION
**Pull request by demon1999**
better command line in Options.cpp

### CODE REVIEW

**Comment by winsvega** [comment]
`test/tools/libtesteth/Options.cpp: In constructor ‘dev::test::Options::Options(int, char**)’:
/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp:115:60: error: expected ‘,’ before ‘.’ token
   auto throwIfNoArgumentFollows = [&i, &unrecognisedOptions.size(), &arg]()
                                                            ^
/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp:115:60: error: expected identifier before ‘.’ token
/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp:115:67: error: expected ‘]’ before ‘,’ token
   auto throwIfNoArgumentFollows = [&i, &unrecognisedOptions.size(), &arg]()
                                                                   ^
/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp: In lambda function:
/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp:115:67: error: expected ‘{’ before ‘,’ token
/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp: In constructor ‘dev::test::Options::Options(int, char**)’:
/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp:115:73: error: expected initializer before ‘]’ token
   auto throwIfNoArgumentFollows = [&i, &unrecognisedOptions.size(), &arg]()
                                                                         ^
/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp:137:16: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
    if (i + 1 < argc) // two params
                ^
/home/wins/Ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp:208:11: error: ‘generate’ is not a member of ‘dev::test::RandomCode’
   cout << dev::test::RandomCode::generate(maxCodes) << "\n";
           ^
cc1plus: all warnings being treated as errors
make[2]: *** [test/CMakeFiles/testeth.dir/tools/libtesteth/Options.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs...`

**Comment by demon1999** [comment]
fixed

**Comment by winsvega** [comment]
hmm. it still shows compiler errors:

```
[ 78%] Building CXX object test/CMakeFiles/testeth.dir/tools/libtesteth/Options.cpp.o
/home/travis/build/ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp: In constructor ‘dev::test::Options::Options(int, char**)’:
/home/travis/build/ethereum/cpp-ethereum/test/tools/libtesteth/Options.cpp:164:47: error: cannot convert ‘__gnu_cxx::__alloc_traits<std::allocator<std::basic_string<char> > >::value_type {aka std::basic_string<char>}’ to ‘const char*’ for argument ‘1’ to ‘int atoi(const char*)’
    trDataIndex = atoi(unrecognisedOptions[++i]); 
```

**Comment by demon1999** [comment]
fixed

File: test/tools/libtesteth/Options.cpp [hunk 1]
```diff
@@ -22,20 +22,22 @@
...
 using namespace std;
 using namespace dev::test;
 using namespace dev::eth;
+namespace po = boost::program_options;
 
 void printHelp()
 {
 	cout << "Usage: \n";
-	cout << std::left;
+	cout << std::left << "\n";
```
Comment (pirapira) [inline test/tools/libtesteth/Options.cpp]: Why?

File: test/tools/libtesteth/Options.cpp [hunk 2]
```diff
@@ -75,12 +77,45 @@ Options::Options(int argc, char** argv)
 	trGasIndex = -1;
 	trValueIndex = -1;
 	bool seenSeparator = false; // true if "--" has been seen.
-	for (auto i = 0; i < argc; ++i)
+	po::options_description testSuits("Setting test suite");
+	testSuits.add_options()
+			("testpath", po::value<string>(), "<PathToTheTestRepo>");
+	po::options_description debugging("Debugging");
+	debugging.add_options()
+			("verbosity", po::value<string>(), "<level> Set logs verbosity. 0 - silent, 1 - only errors, 2 - informative, >2 - detailed")
```
Comment (pirapira) [inline test/tools/libtesteth/Options.cpp]: Now what happens when `--verbosity 6` appears before the separator `--`?

**Comment by winsvega** [comment]
hm.. what happend to the tests? 

```
/home/travis/build/ethereum/cpp-ethereum/test/tools/jsontests/BlockChainTests.cpp(1008): last checkpoint: "bcGasPricerTest" fixture ctor
unknown location(0): fatal error: in "BlockchainTests/bcInvalidHeaderTest": boost::filesystem::filesystem_error: 
boost::filesystem::directory_iterator::construct: No such file or directory: "../../test/jsontests/BlockchainTests/bcInvalidHeaderTest"
```

**Comment by winsvega** [comment]
different padding
```
testeth --help

Setting test suite
-t <TestSuite>                Execute test operations
 -t <TestSuite>/<TestCase>
    --testpath <PathToTheTestRepo>

..

--singletest <TestName>       Run on a single test
    --singletest <TestFile> <TestName>

..

--filltests                   Run test fillers
        --fillchain                   When filling the state tests, fill tests as blockchain instead
```

**Comment by winsvega** [comment]
undefined options works 
```
./testeth -t GeneralStateTests/stExample -- --testpath2sdgds
```

testpath option does not work 

```
./testeth -t GeneralStateTests/stExample -- --testpath "L"
```

**Comment by pirapira** [comment]
@demon1999 please add test cases containing @winsvega 's examples.  It can be a shell script that calls `./testeth` with various valid and invalid options.

**Comment by pirapira** [comment]
@demon1999 is it really recommended to use `boost::test` and `boost::program_options` together?

**Pull request closed without merging**